pkg/daemon/update_test.go

```diff
@@ -579,97 +579,140 @@ func TestDropinCheck(t *testing.T) {
-		"pullsecret2":     helpers.NewIgnFile("/var/lib/kubelet/config.json", "kubelet conf 2\n"),
-		"registries1":     helpers.NewIgnFile("/etc/containers/registries.conf", "registries content 1\n"),
-		"registries2":     helpers.NewIgnFile("/etc/containers/registries.conf", "registries content 2\n"),
-		"randomfile1":     helpers.NewIgnFile("/etc/random-reboot-file", "test\n"),
-		"randomfile2":     helpers.NewIgnFile("/etc/random-reboot-file", "test 2\n"),
-		"kubeletCA1":      helpers.NewIgnFile("/etc/kubernetes/kubelet-ca.crt", "kubeletCA1\n"),
-		"kubeletCA2":      helpers.NewIgnFile("/etc/kubernetes/kubelet-ca.crt", "kubeletCA2\n"),
-		"policy1":         helpers.NewIgnFile("/etc/containers/policy.json", "policy1"),
-		"policy2":         helpers.NewIgnFile("/etc/containers/policy.json", "policy2"),
-		"containers-gpg1": helpers.NewIgnFile("/etc/machine-config-daemon/no-reboot/containers-gpg.pub", "containers-gpg1"),
-		"containers-gpg2": helpers.NewIgnFile("/etc/machine-config-daemon/no-reboot/containers-gpg.pub", "containers-gpg2"),
+		"pullsecret1":              helpers.NewIgnFile("/var/lib/kubelet/config.json", "kubelet conf 1\n"),
+		"pullsecret2":              helpers.NewIgnFile("/var/lib/kubelet/config.json", "kubelet conf 2\n"),
+		"registries1":              helpers.NewIgnFile("/etc/containers/registries.conf", "registries content 1\n"),
+		"registries2":              helpers.NewIgnFile("/etc/containers/registries.conf", "registries content 2\n"),
+		"randomfile1":              helpers.NewIgnFile("/etc/random-reboot-file", "test\n"),
+		"randomfile2":              helpers.NewIgnFile("/etc/random-reboot-file", "test 2\n"),
+		"kubeletCA1":               helpers.NewIgnFile("/etc/kubernetes/kubelet-ca.crt", "kubeletCA1\n"),
+		"kubeletCA2":               helpers.NewIgnFile("/etc/kubernetes/kubelet-ca.crt", "kubeletCA2\n"),
+		"policy1":                  helpers.NewIgnFile("/etc/containers/policy.json", "policy1"),
+		"policy2":                  helpers.NewIgnFile("/etc/containers/policy.json", "policy2"),
+		"containers-gpg1":          helpers.NewIgnFile("/etc/machine-config-daemon/no-reboot/containers-gpg.pub", "containers-gpg1"),
+		"containers-gpg2":          helpers.NewIgnFile("/etc/machine-config-daemon/no-reboot/containers-gpg.pub", "containers-gpg2"),
+		"fcos_authorized_keys":     helpers.NewIgnFile(fcosAuthKeyPath, "authorized_keys"),
+		"non_fcos_authorized_keys": helpers.NewIgnFile(nonFCOSAuthKeyPath, "authorized_keys"),
 	}
 
 	tests := []struct {
 		oldConfig      *mcfgv1.MachineConfig
 		newConfig      *mcfgv1.MachineConfig
+		os             OperatingSystem
 		expectedAction []string
 	}{
 		{
 			// test that a normal file change is reboot
 			oldConfig:      helpers.NewMachineConfig("00-test", nil, "dummy://", []ign3types.File{files["randomfile1"]}),
 			newConfig:      helpers.NewMachineConfig("01-test", nil, "dummy://", []ign3types.File{files["randomfile2"]}),
+			os:             FCOS,
```

Comment:
There's two down at the bottom which I think cover every case. Anything I still need to add?

Reply:
I looked at `osrelease.go` and I don't think there's any values for RHEL. For RHCOS, I'm not sure what VariantID should be. I guess I'd need to look at what os-release contains on those OSes. I just used `OperatingSystem{}` since `IsFCOS` will return false and that's the if branch I was trying to test

Reply:
Matthew updated the tests and this is fixed.
For posterity as per https://github.com/openshift/os/commit/31f295e3362a6622749a64a6ff610b727560bda1 and his manual checking RHCOS doesn't have Variant ID